LEAF_Request_Portal/js/vue_conditions_editor/LEAF_conditions_editor.js

```diff
@@ -0,0 +1,726 @@
+                @set-condition="selectConditionFromList"
+                @remove-condition="removeCondition"
+                @cancel-delete="showRemoveConditionModal=false"
+                @update-selected-child-value="updateSelectedChildValue">
+            </editor-main>
+            <editor-actions
+                :conditionInputComplete="conditionComplete"
+                :parentIndicator="selectedParentIndicator"
+                :childIndicator="childIndicator"
+                :conditions="conditionInputObject"
+                :showRemoveConditionModal="showRemoveConditionModal"
+                @save-condition="postCondition"
+                @cancel-entry="clearSelections(true)">
+            </editor-actions>
+        </div>
+    </div>` 
+});
+
+
+
+//CENTER EDITOR WIDGET
+ConditionsEditor.component('editor-main', {
+    props: {
+        vueData: Object,
+        selectedChild: Object,
+        selectableParents: Array,
+        indicators: Array,
+        selectedParentOperators: Array,       //available operators, based on format of above ind
+        selectedParentValueOptions: Array,    //values for dropdown formats
+        selectedChildValueOptions: Array,
+        parentFormat: String,
+        childFormat: String,
+        conditions: Object,
+        showConditionEditor: Boolean,
+        showRemoveConditionModal: Boolean,
+        editingCondition: String
+    },
+    methods: {
+        validateCurrency(event) {
+            const currencyRegex = /^(\d*)(\.\d{0,2})?$/;
+            const val = event.target.value;
+            if (!currencyRegex.test(val)) { //TODO: userfeedback
+                document.getElementById('currency-format-input').value = '';
+            } else {
+                this.$emit('update-selected-parent-value', event.target.value);
+            }
+        },
+        forceUpdate(){
+            this.$forceUpdate();
+            if(this.vueData.updateIndicatorList === true){ //set to T in mod_form if new ind or ind edited, then to F after new fetch
+                this.$emit('update-indicator-list');
+            } else {
+                this.$emit('update-selected-child');
+                /*if (this.vueData.required === 0) {
+                    this.$emit('update-selected-child');
+                } else {
+                    alert('This question is Required, currently questions that are required are not supported with conditions. ' +
+                        'Please remove the required status and try again.');
+                    this.vueData.indicatorID = 0;
+                }*/
+            }
+        },
+        getIndicatorName(id){
+            let indicatorName = '';
+            indicatorName = this.indicators.find(indicator => indicator.indicatorID === id)?.name;
+            indicatorName = indicatorName.length > 50 ? indicatorName.slice(0,50) + '... ' : indicatorName;
+            return indicatorName;
+        },
+        getOperatorText(op){

```

Comment:
The below function is in the scope of just this one branch at least 4 times.  This should be pulled out to a central location for use instead of repeated code.